propagators/jaeger/jaeger_propagator.go

```diff
@@ -0,0 +1,152 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaeger
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/otel/api/propagation"
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+const (
+	jaegerHeader        = "uber-trace-id"
+	separator           = ":"
+	traceID64bitsWidth  = 64 / 4
+	traceID128bitsWidth = 128 / 4
+	spanIDWidth         = 64 / 4
+
+	traceIDPadding = "0000000000000000"
+
+	flagsDebug      = 0x02
+	flagsSampled    = 0x01
+	flagsNotSampled = 0x00
+
+	deprecatedParentSpanID = "0"
+)
+
+var (
+	empty = trace.EmptySpanContext()
+
+	errMalformedTraceContextVal = errors.New("header value of uber-trace-id should contain four different part separated by : ")
+	errInvalidTraceIDLength     = errors.New("invalid trace id length, must be either 16 or 32")
+	errMalformedTraceID         = errors.New("cannot decode trace id from header, should be a string of hex, lowercase trace id can't be all zero")
+	errInvalidSpanIDLength      = errors.New("invalid span id length, must be 16")
+	errMalformedSpanID          = errors.New("cannot decode span id from header, should be a string of hex, lowercase span id can't be all zero")
+	errMalformedFlag            = errors.New("cannot decode flag")
+)
+
+type Jaeger struct{}
+
+var _ propagation.HTTPPropagator = &Jaeger{}
+
+func (jaeger Jaeger) Inject(ctx context.Context, supplier propagation.HTTPSupplier) {
```

Comment:
Same:

```suggestion
// Inject injects a context into the supplier as Jaeger headers.
// The parent span ID is omitted because it is not tracked in the
// SpanContext.
func (jaeger Jaeger) Inject(ctx context.Context, supplier propagation.HTTPSupplier) {
```